utils/generate_mbox.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+# You can use this script to create a test mbox with specific mail size
+# distribution.
+#
+# Running the script creates a 'testmbox' output file in the current directory.
+
+from email.mime.text import MIMEText
+from email.utils import formatdate
+import mailbox
+import random
+import string
+import os
+
+# Size distribution can be configured here. By default, a mbox with 5 mails of
+# 10kB, 80kB, 150kB and 250kB each in a randomized order are created.
+# ImapTest iterates through the mbox sequentially, so all randomness must be
+# in the mbox file.
+size_distribution = {
+  10:    5,
+  80:    5,
+  150:   5,
+  250:   5,
+  15000: 0,
+}
+
+mailfrom = 'sender@example.com'
+mailto = 'recipient@example.com'
+mbox_out = 'testmbox'
+mbox_tmp = '/tmp/testmbox'
```

Comment:
should we use tempfile.NamedTemporaryFile instead?

Reply:
or tempfile.mkstemp()

Reply:
This code works so should we really trying to be micro-optimizing here?

Reply:
I just wonder if it would be prudent to use a tempfile that's not gonna exist in case something goes bad, it's more of a hygiene thing than micro-optimization.

Reply:
OK, added mkstemp() usage.  Annoying that mailbox has an old API that only takes filenames and not handles.